### PR TITLE
dev-libs/wlc: add postinst message about Xwayland.

### PR DESCRIPTION
**Pull request by tajjada**
This is a cleaner and less confusing separation into USE flags.

This change was already discussed at length in IRC in #gentoo-proxy-maint.

Package-Manager: portage-2.2.28

### CODE REVIEW

File: dev-libs/wlc/wlc-0.0.2-r1.ebuild [hunk 1]
```diff
@@ -28,6 +30,7 @@ RDEPEND="virtual/opengl
 			 x11-libs/xcb-util-image
 			 x11-libs/xcb-util-wm
 			 x11-libs/libXfixes )
+		xwayland? ( x11-base/xorg-server[wayland] )
```
Comment (mgorny) [inline dev-libs/wlc/wlc-0.0.2-r1.ebuild]: Why the dependency? It doesn't seem like the package would build or work incorrectly if people don't have wayland installed.

**Comment by mgorny** [comment]
xwayland or wayland? From what I can see, xwayland is about running native X11 apps on wayland… so it's basically no wayland support at all, just an implicit feature of wayland.

**Comment by tajjada** [comment]
@mgorny 

Yes, correct, it is about the ability to run X11 apps in a wayland environment. However, Xwayland is not a feature of wayland, it is a feature of the compositor.

In wlc, the 'X' use flag enables X11-related code in wlc, which includes a X window rendering backend (to render graphics to a X window, rather than your monitor), and Xwayland support.

To actually use the Xwayland support, however, you need to have x11-base/xorg-server[wayland], which provides the Xwayland binary, which wlc (and other compositors supporting Xwayland, such as weston) uses at runtime. Without this, the Xwayland functionality is unusable/unavailable. This new change introduces a new USE flag that adds this as a dependency, to make Xwayland work.

Even without this change, the wlc ebuild technically supports Xwayland, as the 'X' use flag already enables the needed code. However, it cannot be used without xorg-server[wayland], which users would have to install separately. The new USE flag introduced here makes it easier, more user-friendly, and more intuitive to have everything install correctly and just work.

I have been discussing this change over the past few hours on IRC. Originally, I had an 'X' use flag in my sway ebuild (that I am currently trying to get merged into gentoo), which pulled in this xorg-server[wayland] dependency. However, that made no sense, as the Xwayland functionality is actually implemented by wlc, and not by sway. Hence, it should be wlc that pulls in this dependency.

**Comment by mgorny** [comment]
How likely it is that users is going to want Xwayland in wlc, and not have it enabled in xorg-server? The problem is, by introducing this flag you are effectively causing two issues:

1. people who want Xwayland will rebuild wlc unnecessarily even though nothing actually changes in the build,
2. people who want to get rid of Xwayland will have to rebuild wlc without the USE flag, in order to be able to rebuild xorg-server.

This is considered a QA violation by creating USE flags that don't affect the build, and cause unnecessary rebuilds and/or binary package mismatches without real reason to do so.

**Comment by tajjada** [comment]
@mgorny Thank you for pointing this out. You are entirely correct. I accept that I made a stupid and inexperienced mistake here. I will replace the USE flag with a message instead, and edit this pull request accordingly. Give me a few minutes.

**Comment by mgorny** [comment]
No problem at all. It's all about learning after all.

**Comment by tajjada** [comment]
@mgorny 

Fixed it. The ebuild now displays a postinst message instead. Have a look and tell me if there is still anything I am doing wrong.

File: dev-libs/wlc/wlc-0.0.2-r1.ebuild [hunk 2]
```diff
@@ -48,3 +48,11 @@ src_configure() {
 
 	cmake-utils_src_configure
 }
+
+pkg_postinst() {
+	if use X
```
Comment (mgorny) [inline dev-libs/wlc/wlc-0.0.2-r1.ebuild]: You may also want to use `has_version 'x11-base/xorg-server[wayland]'` to check if xorg-server is already built correctly, and display the message only if it isn't.

**Comment by tajjada** [comment]
Also, before you merge this, could you tell me, is a revision bump actually necessary? It is just a new message, after all. I left it just in case, but maybe I can re-do this commit without the revision bump.

File: dev-libs/wlc/metadata.xml [hunk 3]
```diff
@@ -9,8 +9,4 @@
 		<email>proxy-maint@gentoo.org</email>
 		<name>Proxy Maintainers</name>
 	</maintainer>
-	<use>
-		<flag name="systemd">Enable support for systemd-logind.</flag>
-		<flag name="X">Enable X11 backend and XWayland support.</flag>
```
Comment (mgorny) [inline dev-libs/wlc/metadata.xml]: Are you sure you wanted to remove all flag descriptions? Seems unrelated.

File: dev-libs/wlc/wlc-0.0.2-r1.ebuild (hunk 2 above)
Comment (tajjada) [inline dev-libs/wlc/wlc-0.0.2-r1.ebuild]: @mgorny 

I didn't know I could do that. I'll do it; thanks for the suggestion.

File: dev-libs/wlc/metadata.xml (hunk 3 above)
Comment (tajjada) [inline dev-libs/wlc/metadata.xml]: I was just thinking that the descriptions are unnecessary, because these are standard global use flags.

Yeah, it is an unrelated change. I should probably leave them for now, because it is irrelevant to this pull request.

Comment (mgorny) [inline dev-libs/wlc/metadata.xml]: Just to be clear, it's fine to include multiple changes in a single pull request. However, since we're using git, it would be a good idea to split it into a separate commit. This way it's easier to follow changes, rationale and revert if necessary.

**Comment by idella** [comment]
This amounts to a minor change. I can add / commit this
